### DISCUSSION TITLE
Artifacts Not Working with gptPlugins or agents Endpoint

### DISCUSSION THREAD

**Discussion by ogipogi** [danny-avila/LibreChat#4582] [Troubleshooting] [answered]
### What happened?

We are experiencing an issue where artifacts are not functioning when using the gptPlugins endpoint. The same configuration works correctly with the azureOpenAI endpoint. We have custom plugins that users need to choose, and we would like to add artifacts support for the gptPlugins endpoint.

### Steps to Reproduce

Configuration that work:
librechat.yaml
```endpoints:
  gptPlugins:
    titleConvo: true
    titleModel: "gpt-4o"
    plugins: true
    tools: ["CustomXY", "dalle"]
    groups:
      - group: "XY"
        apiKey: "XY"
        instanceName: "XY"
        models:
          gpt-4o:
            deploymentName: "XY"
            version: "2024-02-15-preview"
  azureOpenAI:
    titleConvo: true
    titleModel: "gpt-4o"
    plugins: true
    groups:
      - group: "XY"
        apiKey: "XY"
        instanceName: "XY"
        models:
          gpt-4o:
            deploymentName: "XY"
            version: "2024-02-15-preview"
modelSpecs:
  enforce: false
  prioritize: true
  list:
    - name: "adc-default-specs"
      label: "GPT-4o"
      description: "XY"
      iconURL: "XY"
      preset:
        default: false
        endpoint: "azureOpenAI" # Artifacts works with this endpoint
        #endpoint: "gptPlugins" # Artifacts don't work with this endpoint
        plugins: true
        model: "gpt-4o"
```

Configuration that don't work:
librechat.yaml
```endpoints:
  gptPlugins:
    titleConvo: true
    titleModel: "gpt-4o"
    plugins: true
    tools: ["CustomXY", "dalle"]
    groups:
      - group: "XY"
        apiKey: "XY"
        instanceName: "XY"
        models:
          gpt-4o:
            deploymentName: "XY"
            version: "2024-02-15-preview"
  azureOpenAI:
    titleConvo: true
    titleModel: "gpt-4o"
    plugins: true
    groups:
      - group: "XY"
        apiKey: "XY"
        instanceName: "XY"
        models:
          gpt-4o:
            deploymentName: "XY"
            version: "2024-02-15-preview"
modelSpecs:
  enforce: false
  prioritize: true
  list:
    - name: "adc-default-specs"
      label: "GPT-4o"
      description: "XY"
      iconURL: "XY"
      preset:
        default: false
        #endpoint: "azureOpenAI" # Artifacts works with this endpoint
        endpoint: "gptPlugins" # Artifacts don't work with this endpoint
        plugins: true
        model: "gpt-4o"
```

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment, marked as answer]
Thanks for your report. It won't be supported. Plugins are deprecated soon in favor of a better system, which will also incorporate artifacts: https://github.com/danny-avila/LibreChat/issues/3607

**Reply by ogipogi** [reply]
Thanks for the fast answer. I had a look at https://github.com/danny-avila/LibreChat/issues/3607. Did I get it right that the new system with agents can already be tested?

**Reply by danny-avila** [reply]
Yes it can already be tested!

**Reply by ogipogi** [reply]
@danny-avila, I tested it successfully. 

I needed to create an agent with the builder UI.

Can I have the agent preconfigured via librechat.yaml?

**Reply by ogipogi** [reply]
Ok, i think i figured it out - with using agent_id in the preset.

**Reply by ogipogi** [reply]
The Artifacts feature is not yet working with the new Agent System. @danny-avila, can you confirm this? Or am I doing something wrong?

**Reply by ogipogi** [reply]
I updated the title by adding also the agents endpoint.

**Reply by danny-avila** [reply]
Agents are not officially released yet, I've marked the current usage as experimental.

**Reply by ogipogi** [reply]
Thanks for your response @danny-avila . We have switched to the agent's system because we need to use our custom plugins, as the plugin system will be soon deprecated.
